Cargo.toml

```diff
@@ -227,3 +230,12 @@ conf-files = [
 
 [package.metadata.spellcheck]
 config = "./scripts/ci/gitlab/spellcheck.toml"
+
+# Make sure xcm-emulator cumuls deps work as otherwise they depend on polkadot master/other branch.
+# This will also override any other polkadot transitive dependencies.
+[patch.'https://github.com/paritytech/polkadot']
+polkadot-core-primitives = { path = "core-primitives" }
+polkadot-parachain = { path = "parachain" }
+xcm = { path = "xcm" }
+xcm-executor = { path = "xcm/xcm-executor" }
+xcm-builder = { path = "xcm/xcm-builder" }
```

Comment:
do we need this?

Reply:
yes, this is in the description and there is also a comment in the Cargo.toml as to why. The unfortunate reality here is that since we need both polkadot and cumulus as dependencies for xcm-emulator - we also need to make sure that transitive dependencies are all the same versions. And that does not seem achievable otherwise, neither in polkadot nor in cumulus. It would be if we had a monorepo.